Frontend-awas

Awasjo opened the pull request:
Added login, registration and dashboard 
UI is bad right now, but will be fixed with later commits

Awasjo committed:
feat: Add api services and screens

Awasjo committed:
fix: fixed registration error changed Username to Name to back database field

kentayamada-centennial commented on frontend/screens/DashboardScreen.js:
```diff
@@ -0,0 +1,48 @@
+  }, [userId]);
+
+  return (
+    <div>
```
Using h1, h2, and other HTML heading tags directly in React Native code is not applicable because React Native is designed for building native mobile applications, and it uses its own set of components rather than HTML elements.

https://reactnative.dev/docs/components-and-apis

kentayamada-centennial commented on backend/api/Program.cs:
```diff
@@ -6,6 +6,13 @@
 
+builder.Services.AddCors(options => {
+    options.AddPolicy("AllowSpecificOrigin", builder =>
+        builder.WithOrigins("http://localhost:8081") 
```
Using an environment variable for the origin URL is a good practice because it makes your code more flexible and secure, especially when moving between different environments (development, staging, production).

Awasjo replied:
alright, i will fix this now

Awasjo committed:
security: updated program.cs for user's localhostURL

changed it so that the localhost url comes from the env of the user's settings similar to connection string

Awasjo committed:
pending.DashboardScreenSummary commit to work on laptop

Awasjo committed:
fix: dashboard screen flat list

Awasjo committed:
updated back end to match kenta

Awasjo committed:
feat: added account screen and create account screen
tomorrow: will add edit screen
